disconnected_registry/docker-registry-sync.py

```diff
@@ -29,6 +29,9 @@
                                                                          'dumped to stdout instead of run')
 parser.add_argument('--openshift-version', action='store', dest='ocp_version', help='The version of OpenShift which you '
                                                                               'want to sync images for')
+parser.add_argument('--remote-registry-user', action='store', dest='remote_registry_user', help='The user for source registry')
+parser.add_argument('--remote-registry-pass', action='store', dest='remote_registry_pass', help='The password for source registry, e.g user token for registry.redhat.io')
+parser.add_argument('--remote-registry-auth-url', action='store', dest='remote_registry_auth_url', help='The registry WWW-authenticate auth challenge URL')
```

Comment:
The username and password should be abstracted from the user when running this command. These values should be retrieved from the `docker.json` file for which the user would have needed to be logged into prior to running the script

The auth challenge URL is returned by the `www-authenticate` header upon a 401. I would recommend that we instead parse the header value instead of putting the onus on the user

Reply:
If i understand it correctly,
if `session.status_code == 401`: then
1. read ` www-authenticate` challenge value, exit when no `www-authenticate` not in the header with general exception. 
2. Attempt to validate user from information in the docker.json, exit when failed to read.

As guidance:
1. no login parameter should be part of the script input parameters.
2. User need to do docker login first and we read these information from docker.json created with the logged in user.